src/libcore/cell.rs

```diff
@@ -570,11 +570,17 @@ impl Display for BorrowMutError {
     }
 }
 
-// Values [1, MAX-1] represent the number of `Ref` active
-// (will not outgrow its range since `usize` is the size of the address space)
+// Values [1, MIN_WRITING-1] represent the number of `Ref` active. Values in
+// [MIN_WRITING, MAX-1] represent the number of `RefMut` active. Multiple
+// `RefMut`s can only be active at a time if they refer to distinct,
+// nonoverlapping components of a `RefCell` (e.g., different ranges of a slice).
+//
+// `Ref` and `RefMut` are both two words in size, and so there can never be
+// enough `Ref`s or `RefMut`s in existence to overflow half of the `usize`
+// range. Thus, a `BorrowFlag` will never overflow.
```

Comment:
I doubt this, since you could `forget` a Ref. See #25841.

Reply:
Good call. I'll make sure that this is `assert`ed everywhere it could overflow.

Reply:
OK, I believe I've gotten all of the cases.